shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -812,4 +813,22 @@ std::optional<LRESULT> FlutterWindowsEngine::ProcessExternalWindowMessage(
   return std::nullopt;
 }
 
+bool FlutterWindowsEngine::SendAppLifecycleStateUpdate(
+    AppLifecycleState state) {
+  if (!app_lifecycle_enabled_) {
```

Comment:
What's the benefit of storing `app_lifecycle_enabled_` state in `FlutterWindowsEngine`?

Would it be preferrable to store it in the `WindowsLifecycleManager`? That already stores whether to handle application exits.